data-access/mandataris-download.ts

```diff
@@ -0,0 +1,261 @@
+      BIND(IF(BOUND(?einde), ?einde, ${escapedTodaysDate}) AS ?saveEinde).
+    `;
+  }
+
+  if (persoonIds.length >= 1) {
+    const idValues = persoonIds.map((id) => sparqlEscapeString(id)).join('\n');
+    persoonFilter = `
+      VALUES ?persoonId { ${idValues} }
+
+      ?mandataris mandaat:isBestuurlijkeAliasVan ?persoon.
+      ?persoon mu:uuid ?persoonId.
+    `;
+  }
+
+  if (bestuursFunctieCodeIds.length >= 1) {
+    const idValues = bestuursFunctieCodeIds
+      .map((id) => sparqlEscapeString(id))
+      .join('\n');
+    mandaatTypeFilter = `
+      VALUES ?functieCode { ${idValues} }
+      ?mandaat org:role ?bestuursfunctieCode .
+      ?bestuursfunctieCode mu:uuid ?functieCode.
+    `;
+  }
+
+  const queryString = `
+    PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
+    PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+    PREFIX lmb: <http://lblod.data.gift/vocabularies/lmb/>
+    PREFIX org: <http://www.w3.org/ns/org#>
+    PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+    PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+    PREFIX fractieType: <http://data.vlaanderen.be/id/concept/Fractietype/>
+
+
+    SELECT DISTINCT ?mandataris
+    WHERE {
+      ?mandataris a mandaat:Mandataris.
+      ?mandataris org:holds ?mandaat.
+
+      ?bestuursorgaan lmb:heeftBestuursperiode ?bestuursperiode.
+      ?bestuursperiode mu:uuid ${sparqlEscapeString(bestuursperiodeId)}.
+
+      ${bestuursorgaanInTijdFilter ?? ''}
+      ${onlyActiveFilter ?? ''}
+      ${persoonFilter ?? ''}
+      ${createFractieFilter(filters)}
+      ${mandaatTypeFilter ?? ''}
+    }
+  `;
+
+  const sparqlResult = await query(queryString);
+
+  return getSparqlResults(sparqlResult).map((res) => res.mandataris.value);
+}
+
+function createFractieFilter(filters): string {
+  const {
+    fractieIds,
+    hasFilterOnOnafhankelijkeFractie,
+    hasFilterOnNietBeschikbareFractie,
+  } = filters;
+
+  if (
+    fractieIds.length === 0 &&
+    !hasFilterOnNietBeschikbareFractie &&
+    !hasFilterOnOnafhankelijkeFractie
+  ) {
+    return '';
+  }
+
+  const idValues = fractieIds.map((id) => sparqlEscapeString(id)).join('\n');
+  const unions = [
+    {
+      apply: fractieIds.length >= 1,
+      filter: `
+        UNION {
+          VALUES ?fractieId {\n ${idValues} }
+          ?fractie mu:uuid ?fractieId.
+        }
+      `,
+    },
+    {
+      apply: hasFilterOnNietBeschikbareFractie,
+      filter: `
+        UNION {
+         FILTER NOT EXISTS {
+           ?mandataris org:hasMembership ?lidmaatschap.
+           ?lidmaatschap org:organisation ?fractie.
+         }
+        }
+      `,
+    },
+    {
+      apply: hasFilterOnOnafhankelijkeFractie,
+      filter: `
+        UNION {
+         ?fractie ext:isFractietype fractieType:Onafhankelijk.
+        }
+      `,
+    },
+  ];
+  const unionFilters = unions
+    .filter((union) => union.apply)
+    .map((union) => union.filter)
+    .join('\n');
```

Comment:
this won't look right because your first statement will start with UNION as well. the UNION bit should be removed from the filter statements and added to the join statement

Reply:
When I add it to the join it won't add the  union to the first one, thats expected  right? SO i have to add the union in the filter.

This is the output with join of `\n UNION`

```
mandataris-1  |     {
mandataris-1  |       {
mandataris-1  |       ?mandataris org:hasMembership ?lidmaatschap.
mandataris-1  |         ?lidmaatschap org:organisation ?fractie.
mandataris-1  |       }
mandataris-1  |         {
mandataris-1  |           VALUES ?fractieId {
mandataris-1  |  """5C3DBE9C57753A00090001E3""" }
mandataris-1  |           ?fractie mu:uuid ?fractieId.
mandataris-1  |         }
mandataris-1  |       
mandataris-1  |  UNION 
mandataris-1  |         {
mandataris-1  |          FILTER NOT EXISTS {
mandataris-1  |            ?mandataris org:hasMembership ?lidmaatschap.
mandataris-1  |            ?lidmaatschap org:organisation ?fractie.
mandataris-1  |          }
mandataris-1  |         }
mandataris-1  |       
mandataris-1  |  UNION 
mandataris-1  |         {
mandataris-1  |          ?fractie ext:isFractietype fractieType:Onafhankelijk.
mandataris-1  |         }
mandataris-1  |       
mandataris-1  |     }
mandataris-1  |   
mandataris-1  |       
mandataris-1  |     }

Reply:
so as discussed, this part needs to be removed: ```  
  {
mandataris-1  |       ?mandataris org:hasMembership ?lidmaatschap.
mandataris-1  |         ?lidmaatschap org:organisation ?fractie.
mandataris-1  |       }
```

Reply:
Updated